src/game/aggregation.rs

```diff
@@ -0,0 +1,561 @@
+    ) {
+        // Generally, game needs to be reverted to this history after playing any action
+        // This is necessary to process data for multiple lines over the game
+        let history = game.history().to_owned();
+
+        if game.is_chance_node() {
+            // If we're at a chance node, call `update_report_for_game` on self for each possible chance card
+            for card in 0..52 {
+                // Don't process chance cards that are already on the board
+                if game.current_board().contains(&card) {
+                    continue;
+                }
+
+                game.play(card as usize);
+
+                self.update_report_for_game_with_frequency(game, global_frequency);
+
+                game.back_to_root();
+                game.apply_history(&history);
+            }
+        } else {
+            // Otherwise, update the data, then recursively update all child nodes
+            game.cache_normalized_weights();
+
+            // Compute statistics
+            let (oop_equity, oop_ev, oop_eqr) = get_player_stats(game, 0);
+            let (ip_equity, ip_ev, ip_eqr) = get_player_stats(game, 1);
+            let board = game.current_board();
+            let action_frequencies = get_action_frequencies(game);
+            let row = AggRow {
+                flop: board[0..3]
+                    .try_into()
+                    .expect("board must at least have flop"),
+                turn: board.get(3).copied(),
+                river: board.get(4).copied(),
+                global_frequency,
+                ip_equity,
+                ip_ev,
+                ip_eqr,
+                oop_equity,
+                oop_ev,
+                oop_eqr,
+                action_frequencies: action_frequencies.clone(),
+                action_evs: get_action_evs(game),
+            };
+            self.data.push(row);
+
+            // Update child nodes for each action
+            // NOTE: actions should only appear in the tree if the line was explicitly requested
+            for (&action, child_tree) in &mut self.child_trees {
+                let action_index = game
+                    .available_actions()
+                    .iter()
+                    .position(|&a| a == action)
+                    .expect("Input PostFlopGame does not match TreeConfig from initialization.");
+
+                game.play(action_index);
+
+                let new_global_frequency = global_frequency * action_frequencies[action_index];
+                child_tree.update_report_for_game_with_frequency(game, new_global_frequency);
+
+                game.back_to_root();
+                game.apply_history(&history);
+            }
+        }
+    }
+}
+
+// NOTE: This is only used by tests and doctests
+// But we can't use cfg(test), cfg(doctest) etc. because
+// cfg(doctest) currently does not work as expected
+/// ONLY USE FOR TESTING
+pub fn load_test_game_and_config() -> (PostFlopGame, TreeConfig) {
```

Comment:
@bkushigian lmk if you have a better solution here. I'd like to use this for both doc tests and actual tests, but apparently this is known to be difficult.

Reply:
Fine for now, we can fix it when `cfg(doctest)` works as expected